### ISSUE TITLE
Can't upgrade to axios 0.26.x without breaking CI

### ISSUE THREAD

**Issue by 0xApotheosis** [shapeshift/web#1326]
## Description

Upgrading axios from `0.24.x` causes non-deterministic failures in Cypress CI.

The 2 errors that occur non-deterministically are:

### 1. Error: ESOCKETTIMEDOUT

https://github.com/shapeshift/web/runs/5699857273?check_suite_focus=true

```
Running:  dashboard_spec.ts                                                               (3 of 3)
  Estimated: 1 minute, 28 seconds
[2424:0326/001916.[134](https://github.com/shapeshift/web/runs/5699857273?check_suite_focus=true#step:8:134)082:ERROR:bus.cc(392)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")


  The Dashboard
    1) "before all" hook for "nav bar works"


  0 passing (31s)
  1 failing

  1) The Dashboard
       "before all" hook for "nav bar works":
     CypressError: `cy.visit()` failed trying to load:

http://localhost:3000/

We attempted to make an http request to this URL but the request failed without a response.

We received this error at the network level:

  > Error: ESOCKETTIMEDOUT
```

### 2. Expected to find element: `[data-test=account-row]`, but never found it.

https://github.com/shapeshift/web/runs/5700651405?check_suite_focus=true

```
Running:  dashboard_spec.ts                                                               (1 of 3)
  Estimated: 1 minute, 11 seconds


  The Dashboard
    1) nav bar works
    ✓ displays the expected account rows (6454ms)
    ✓ supports trades (20[128](https://github.com/shapeshift/web/runs/5700651405?check_suite_focus=true#step:8:128)ms)
    - supports send transaction setup
    - supports receive transaction setup


  2 passing (2m)
  2 pending
  1 failing

  1) The Dashboard
       nav bar works:
     AssertionError: Timed out retrying after 25000ms: Expected to find element: `[data-test=account-row]`, but never found it.
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/dashboard_spec.ts:112:36)
```

## Attempts to fix

1. https://github.com/shapeshift/web/pull/1319
2. https://github.com/shapeshift/web/pull/1299
3. https://github.com/shapeshift/web/pull/1292
4. https://github.com/shapeshift/web/pull/1332
5. https://github.com/shapeshift/web/pull/1417

## Other related PRs

1. https://github.com/shapeshift/web/pull/1275
2. https://github.com/shapeshift/web/pull/1297

## Related Cypress issues

1. https://github.com/cypress-io/cypress/issues/7062

**Comment by GMSteuart** [comment]
@0xApotheosis There are multiple problems that were causing the upgrade to 0.26.x to fail:

- The cypress tests aren't waiting for network requests to finish (default timeout for network requests is 5s)
- The application is making more network requests than needed. I haven't looked at the implementation detail but there needs to be some form of debouncing and cancelling network requests that are no longer needed.
- Passing the options object with a custom timeout to cypress methods that don't have a timeout setting
- Using the cypress `click` method in a chain of commands that has an assertion that wants to naturally retry. These are mutually exclusive as stated by the docs since the click method is an only once method while most other methods keep retrying until the method passes.

I can see a number of new issues being created given the problems stated. So far I can see the following needing to be done:
- Minimize network requests throughout application
- Refactor cypress tests a little and comment where "waiting" needs to occur. The code and comment explanations will serve as a better example when looking at existing code as a reference.
- Add to the docs some of the caveats of cypress, how to go about structuring tests, and when to and not to chain certain commands.
- Aaaaaaand lastly, the one that's already well known....intercepting the request and returning fixture data. Might be easiest to create fixture directions and files that closely mirror the api request with the minimum amount needed to prevent collision (e.g. generalized /host/action/method.key1.key2.ts) 😄 

Link to related pr: pull #1525

**Comment by GMSteuart** [comment]
Related issue #1461

**Comment by GMSteuart** [comment]
@0xApotheosis this is good to go as part of pull #1525
